kvdb/postgres/readwrite_bucket.go

```diff
@@ -427,3 +427,36 @@ func (b *readWriteBucket) Sequence() uint64 {
 
 	return uint64(seq)
 }
+
+// Prefetch will attempt to prefetch all values under a path from the passed
+// bucket.
+func (b *readWriteBucket) Prefetch(paths ...[]string) {}
+
+// ForAll is an optimized version of ForEach with the limitation that no
+// additional queries can be executed within the callback.
+func (b *readWriteBucket) ForAll(cb func(k, v []byte) error) error {
+	rows, cancel, err := b.tx.Query(
+		"SELECT key, value FROM " + b.table + " WHERE " +
+			parentSelector(b.id) + " ORDER BY key",
+	)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+
+	for rows.Next() {
+		var key, value []byte
```

Comment:
Declare at the top of this loop, with the assumption that the caller won't use the byte slices outside of the scope of the closure? A similar assumption exists w.r.t the way bbolt works.

Reply:
For `ForEach`, the bbolt docs say:

>Please note that keys and values in ForEach() are only valid while the transaction is open. If you need to use a key or value outside of the transaction, you must use copy() to copy it to another byte slice.

So the scope for this is wider than just the callback?